File: Codeforces/B_Array_Fix.py
def first(n):
    return n // 10

def second(n):
    return n % 10

def solve():
    n = int(input())
    A = list(map(int, input().split()))
    V = []
    a = first(A[0])
    b = second(A[0])
    if b >= a:
        V.append(a)
        V.append(b)
    else:
        V.append(A[0])
    flag = False
    for i in range(1, n):
        m = first(A[i])
        s = second(A[i])
        if m >= V[-1] and s >= m:
            V.append(m)
            V.append(s)
        else:
            if A[i] >= V[-1]:
                V.append(A[i])
            else:
                flag = True
                break
    print("NO" if flag else "YES")

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        solve()

File: Codeforces/B_Array_Fix.cpp
#include <bits/stdc++.h>
//#define int long long
using namespace std;

void solve(){
int N; cin >> N; vector<int> A(N); for(auto &a : A) cin >> a;
int current = 0;
for(int i = 0 ; i < N ; ++i){
    if(A[i] < current){
        cout << "NO" << endl;
        return;
    }
    else{
        int first = A[i]/10, second = A[i]%10;
        if(first >= current){
            if(second >= first){
                current = second;
            }
            else{
                current = A[i];
            }
        }
        else{
            current = A[i];
        }
    }
}
cout << "YES" << endl;
}

int32_t main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}